src/Table/generate_property_tree/add_to_property_tree.cxx

```diff
@@ -35,18 +35,23 @@ void add_to_property_tree(const Column &column, const std::string &tree_name,
     std::string datatype = to_xml_string(active_datatype);
     field.add("<xmlattr>.datatype", datatype);
 
-    if (active_datatype == Data_Type::CHAR || column.array_size != 1) {
+    bool added_arraysize =
+            (active_datatype == Data_Type::CHAR || column.array_size != 1);
+    if (added_arraysize) {
         field.add("<xmlattr>.arraysize", "*");
     }
     for (auto &a : column.field_properties.attributes) {
-        /// Empty attributes cause field.add to crash :(, so make sure
-        /// that does not happen.
-
+        // Empty attributes cause field.add to crash :(, so make sure
+        // that does not happen.
         // FIXME: This error is thrown a bit too late to be useful.
-
         if (a.first.empty())
             throw std::runtime_error("Empty attribute in field " + column.name +
                                      " which has type " + to_string(column.type));
+
+        if (added_arraysize && boost::equals(a.first, "arraysize")) {
+            continue;
+        }
+
```

Comment:
(This seems unrelated to accessors.) Do I understand correctly that user specified arraysizes will now always be overwritten with `*` for string columns? Why the change?

Reply:
I thought of putting in a comment with a reference to whatever it was that made me do this, but it was so easy to find such references that I didn't bother.  Now of course I can't find any.  I don't remember whether it was an external test that was failing or a warning in an IVOA spec.  I'll see if I can find it and if I can't I'll back this out.

Reply:
The VOTable spec says:
```
Strings, which are defined as a set of characters, can therefore be represented in VOTable as a fixed- or variable- length array of characters...
```
If `*`s are needed for some specific `<FIELD>`s, then maybe fixing the arraysize attributes for those specific fields would be better?

Reply:
On the other hand, tablator doesn't support unbounded length strings, and `*` doesn't seem like it is ever actually wrong, so I am OK with this change. (I'm still curious what exactly motivated this though.)

Reply:
Yes, I certainly wouldn't have dreamt it up myself so *something* must have triggered it.  I'll see if I can track down the reference after I make the other changes.  At the least I could add that attribute only when it isn't already present.

Reply:
Ah, right.  These lines `  if (column.type == Data_Type::CHAR || column.array_size != 1)
    field.add ("<xmlattr>.arraysize", "*");`
have been there preceding the loop through attributes since before my time.  I just added a check during the loop so that we don't wind up with two "arraysize" attributes if the user happened to supply one.

Reply:
I see - thanks for clarifying!